Essentials/src/main/resources/config.yml

```diff
@@ -524,6 +524,13 @@ allow-silent-join-quit: false
 custom-join-message: "none"
 custom-quit-message: "none"
 
+# This message will be shown if a user has joined with a new username
+# This only shows up if they have joined before, but they have since changed their username
+# This will be displayed INSTEAD OF custom-join-message, so if you intend to keep them similar, make sure they match
+# Additionally, this only shows up if a custom-join-message is set, so set something!
+# All custom-x-message placeholders are supported here, and you can use {OLDUSERNAME} to indicate the old username.
```

Comment:
```suggestion
# You can set a custom join message for users who join with a new username here.
# This message will only be used if a user has joined before and have since changed their username.
# This will be displayed INSTEAD OF custom-join-message, so if you intend to keep them similar, make sure they match.
# Set this to "none" to use the the "custom-join-message" above for every join.

# Available placeholders:
# {PLAYER} - The player's displayname. 
# {USERNAME} - The player's username.
# {OLDUSERNAME} - The player's old username.
# {PREFIX} - The player's prefix.
# {SUFFIX} - The player's suffix.
# {ONLINE} - The number of players online.
# {UNIQUE} - The number of unique players to join the server.
# {UPTIME} - The amount of time the server has been online.
```